pkg/infra/infra.go

```diff
@@ -165,20 +166,21 @@ var _ = Describe("Managed cluster should", framework.LabelMachines, func() {
 		client, err = framework.LoadClient()
 		Expect(err).ToNot(HaveOccurred(), "Controller-runtime client should be able to be created")
 
+		DeferCleanup(func() {
+			if machineSet != nil {
+				By("Deleting the new MachineSet")
+				Expect(client.Delete(ctx, machineSet)).To(Succeed(), "MachineSet should be able to be deleted")
+				framework.WaitForMachineSetsDeleted(ctx, client, machineSet)
+			}
+		})
+
```

Comment:
This still has a chance to error. `machineSet` is not cleared by the before each, so the `machineSet` could come from a previous test run and be non-nil. Will a delete call fail if the resource doesn't exist? I expect so.

Wouldn't changing this to the below make it safer?
```suggestion
		{
			// Reset the machineSet between each test
			machineSet = nil
			
			// Make sure to clean up the machineSet, if we create one.
			DeferCleanup(func() {
				if machineSet != nil {
					By("Deleting the new MachineSet")
					Expect(client.Delete(ctx, machineSet)).To(Succeed(), "MachineSet should be able to be deleted")
					framework.WaitForMachineSetsDeleted(ctx, client, machineSet)
				}
			}
		})

```